2/contrib/s2i/run

```diff
@@ -170,6 +175,51 @@ new_password_hash=`obfuscate_password ${JENKINS_PASSWORD:-password} $old_salt`
 mkdir ${JENKINS_HOME}/logs
 ln -sf ${JENKINS_HOME}/logs /var/log/jenkins
 
+# clean up any plugins in JENKINS_HOME/plugins which we previously linked, but
+# have since deleted from the image
+for FILENAME in ${JENKINS_HOME}/plugins/*.jpi; do
+    # test if it is a sym link, otherwise users have overriden, so leave alone
+    # test command silent, don't need dev/null
+    test -h $FILENAME
+    if [ $? -eq 0 ]; then
+	stat -L $FILENAME >& /dev/null
+	if [ $? -eq 0 ]; then
+	    continue
+	fi
```

Comment:
(fwiw I think all the lines above condense down to `if [ ! -e "$FILENAME" ]; then`.  Am not requiring the change be made.)